### DISCUSSION TITLE
blog/p/top-10-code-smells-to-identify-in-pull-requests-with-code-examples

### DISCUSSION THREAD

**Discussion by giscus** [axolo-co/blog.axolo#93] [Announcements]
# blog/p/top-10-code-smells-to-identify-in-pull-requests-with-code-examples

Learn about the top 10 code smells to watch out for in pull requests, with examples and tips for fixing them. Improve the quality of your code and avoid common coding mistakes with these practical tips.

https://axolo.co/blog/p/top-10-code-smells-to-identify-in-pull-requests-with-code-examples

**Comment by nathom** [comment]
For number 4, can't you just do

```javascript
function calculateNetPay(employee) {
    return calculateGrossPay(employee) - employee.taxRate * netPay;
}
```

instead of writing 3 functions? Also, the "un-smelled" version to calculate net pay is incorrect. Consider an employee who worked 80 hours @ $10/hr, 0% tax.

His correct net pay would be `40*10 + 40*15=1000`. Your function would return `40*15 + 80*10 = 1400`

**Comment by NotoriousPyro** [comment]
8 is not correct.

8 only applies if the class was __implements__ and not `extends` because then it would have to implement all the methods of the class it is implementing.

If you used `implements Bird` it would have to define `fly` but since it is extending it, it's a subclass of Bird and inherits the base's methods if not defined in the subclass.

If ostrich extends animal and pigeon extends bird, and bird extends animal, then ostrich is not a substitutable where bird is specified.
